ionicons/IonIcons.m

```diff
@@ -19,10 +19,33 @@ @implementation IonIcons
 + (UIFont*)fontWithSize:(CGFloat)size;
 {
     UIFont* font = [UIFont fontWithName:@"ionicons" size:size];
+    if (!font) {
+        [[self class] dynamicallyLoadFontNamed:@"ionicons"];
```

Comment:
I might still like the `dispatch_once` idea anyway because I think it expresses the process better:

- `dynamicallyLoadFontName` is the first thing that the method does, and it's clear that it will only ever happen once since it's in `dispatch_once`, so that removes some ambiguity and provides an explicit answer to the question 'under what conditions will control flow come through here?'.
- The `font` variable is only ever initialized once and then isn't overwritten.

@maxcampolo interested on your thoughts with this. (I also rebased the podspec changes into a single commit)
https://github.com/sweetmandm/ionicons-iOS/commit/8f692c4961610708f1035de335a81413e5203a69